src/util/misc/objectTransforms.ts

```diff
@@ -60,13 +60,13 @@ export const applyTransformToObject = (
   object: FabricObject,
   transform: TMat2D
 ) => {
-  const { translateX, translateY, scaleX, scaleY, ...otherOptions } =
-      qrDecompose(transform),
+  const { translateX, translateY, ...otherOptions } = qrDecompose(transform),
     center = new Point(translateX, translateY);
-  object.flipX = false;
-  object.flipY = false;
-  Object.assign(object, otherOptions);
-  object.set({ scaleX, scaleY });
+  object.set({
+    flipX: false,
+    flipY: false,
+    ...otherOptions,
+  });
```

Comment:
thos are made explicitly to run set only on necessary side effects, that should be as little as possible.
The only we need is scale, that reset flip or enabe flip.
This shouldn't be the norm, so please don't do it